resources/assets/lib/beatmap-discussions/editor-discussion-component.tsx

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import * as React from 'react';
+import { Path, Transforms } from 'slate';
+import { RenderElementProps } from 'slate-react';
+import { ReactEditor } from 'slate-react';
+import EditorBeatmapSelector from './editor-beatmap-selector';
+import EditorIssueTypeSelector from './editor-issue-type-selector';
+import { SlateContext } from './slate-context';
+
+interface Props extends RenderElementProps {
+  // attributes taken from RenderElementProps, but extended with contentEditable
+  attributes: {
+    contentEditable?: boolean;
+    'data-slate-inline'?: true;
+    'data-slate-node': 'element';
+    'data-slate-void'?: true;
+    dir?: 'rtl';
+    ref: any;
+  };
+  beatmaps: Beatmap[];
+  beatmapset: Beatmapset;
+  currentBeatmap: Beatmap;
+  currentDiscussions: BeatmapDiscussion[];
+  discussionId?: number;
+  editMode?: boolean;
+  readOnly?: boolean;
+}
+
+export default class EditorDiscussionComponent extends React.Component<Props> {
+  static contextType = SlateContext;
+
+  componentDidMount = () => {
+    // reset timestamp to null on clone
+    Transforms.setNodes(this.context, {timestamp: null}, {at: this.path()});
+  }
+
+  componentDidUpdate = () => {
+    const path = this.path();
+
+    if (this.props.element.beatmapId !== 'all') {
```

Comment:
is this the part that's supposed to handle this? Except it doesn't work and gives out "timestamp is specified" error when posted.

(`beatmapId` is undefined for general all)

![](https://s.myconan.net/2020-05/firefox_2020-05-26_21-10-19.png)